### ISSUE TITLE
Adding lsp_message_listener("all") prevents server messages from being sent to the client

### ISSUE THREAD

**Issue by krassowski** [krassowski/jupyterlab-lsp#111]
## Description

Using the example from the docs, all the intercepted messages are from the client. Frontend stops receiving server messages (including the very first response to initialise request) 

## Reproduce

In `load_jupyter_server_extension` of `jupyter_lsp/serverextension.py` add:

```python
    lsp_message_listener = LanguageServerManager.register_message_listener

    @lsp_message_listener("all")
    async def my_listener(scope, message, languages, manager):
        print("received a {} {} message about {}".format(
            scope, message["method"], languages
        ))
```

Observe no server messages being printed out and LSP features no longer working in the frontend. Remove the listener and everything is back to normal.

If we change:
 - `scope='client'` everything seems to work fine.
 - scope='server' leads to the same bug, but no prints (but I see that the client messages go to the server in the server logs)

Note to self: remember to manually restart the lab after changes, the watch does not update python changes.

## Context

- Operating System and version:
- Browser and version: unrelated
- JupyterLab version: 1.2.3
- `jupyter-lsp` version: master
- `@krassowski/jupyterlab-lsp` version: master

<details><summary>Troubleshoot Output</summary>
<pre>
Paste the output from running `jupyter troubleshoot` from the command line here.
You may want to sanitize the paths in the output.
</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
Paste the output from your command line running `jupyter lab` here, use `--debug` if possible.
</pre>
</details>

<details><summary>Browser Output</summary>
<pre>
Paste the output from your browser Javascript console here.
</pre>
</details>

## Coordination

@bollwyvl I will try to write a test for this tonight (London time) but feel free to have your own look.

**Comment by bollwyvl** [comment]
Oh no! Perhaps I missed an await somewhere...

**Comment by bollwyvl** [comment]
Nah, it's that not every message has a `method`, because some are replies. Makes sense! opened up #114, probably ready to go if all the robots are appeased...

**Comment by krassowski** [comment]
Closed by #114
